samples/snippets/src/test/java/beta/video/StreamingAutoMlObjectTrackingIT.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package beta.video;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Integration (system) tests for {@link StreamingAutoMlObjectTracking}. */
+@RunWith(JUnit4.class)
+@SuppressWarnings("checkstyle:abbreviationaswordinname")
+public class StreamingAutoMlObjectTrackingIT {
+
+  private static String PROJECT_ID = "779844219229"; // System.getenv().get("GOOGLE_CLOUD_PROJECT");
+  /** TODO **/
+  private static String MODEL_ID = "VOT3460319223262216192";
+
+  private ByteArrayOutputStream bout;
+  private PrintStream out;
+  private PrintStream originalPrintStream;
+
+  @Before
+  public void setUp() {
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    originalPrintStream = System.out;
+    System.setOut(out);
+  }
+
+  @After
+  public void tearDown() {
+    // restores print statements in the original method
+    System.out.flush();
+    System.setOut(originalPrintStream);
+  }
+
+  @Test
+  public void testStreamingAutoMlObjectTracking() {
+    // Bad Gateway sporadically occurs
```

Comment:
Does `Bad Gateway` happen regularly? If so, is that a problem we should look into fixing on the API side or address in the client library code? We shouldn't expect users to write their own retry logic if they don't have to. This is something client library should hopefully handle.

If this does not happen regularly, we should remove this retry logic.

Reply:
Totally agree! Though, there are some observed unstable behavior on the service side, and while the issues are being surfaced and resolved, we are applying the retry logic for testing the samples. The "Bad Gateway" retry logic is added only to the sample test, so the sample itself on the doc site have clean content.